src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/Internal/DescriptorRegistry.cs

```diff
@@ -1,18 +1,30 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Concurrent;
 using Google.Protobuf.Reflection;
+using Google.Rpc;
 
 namespace Microsoft.AspNetCore.Grpc.JsonTranscoding.Internal;
 
 internal sealed class DescriptorRegistry
 {
+    private readonly object _lock = new object();
     private readonly HashSet<FileDescriptor> _fileDescriptors = new HashSet<FileDescriptor>();
-    private readonly HashSet<EnumDescriptor> _enumDescriptors = new HashSet<EnumDescriptor>();
+    private readonly ConcurrentDictionary<Type, DescriptorBase> _typeDescriptorMap = new ConcurrentDictionary<Type, DescriptorBase>();
```

Comment:
Does this mean that a type can't have both an enum descriptor and message descriptor?

Reply:
That's right.